tests/jballerina-unit-test/src/test/resources/test-src/jvm/type-test-expr.bal

```diff
@@ -170,6 +170,51 @@ function testIsLikeForTupleWithOutRestDescriptor() {
     assertEquals(k4 is [int, string], true);
 }
 
+type R1 readonly & record {|int i;|}?;
+type R2 (readonly & record {|int i;|})?;
+type R record {|int i;|};
+
+function testR1(R1 r) returns boolean {
+    return r is record {|int i;|};
+}
+
+function testR2(R2 r) returns boolean {
+    return r is record {|int i;|};
+}
+
+function testR3(R1 r) {
+    if r is R {
+        R _ = r; // no error
+    } else {
+        assertTrue(false);
+    }
+}
+
+function testR4(R2 r) {
+    if r is R {
+        R _ = r; // no error
+    } else {
+        assertTrue(false);
+    }
+}
```

Comment:
These are "type guard" tests, rather than type test tests.

Reply:
This change is done as per this suggestion https://github.com/ballerina-platform/ballerina-lang/pull/35188#discussion_r851825636

Reply:
Changed